feat: enable nixosTest on aarch64 darwin

jfroche opened the pull request:
Build nixosTest using a local aarch64 linux builder and run nixosTest qemu VM in
aarch64 darwin. This keeps the tests fast as they don't require nested virtualisation.

For example, you can run the nixos test for the http extension on your aarch64 macOS with:

```shell
nix build .\#checks.aarch64-darwin.ext-http
```

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Refactor**
  * Standardized Linux-specific package resolution and host-platform inheritance across test modules and packaging paths.

* **Tests**
  * Unified test harness to a Linux-focused runner, removed host→guest port forwarding and hostPkgs, added /lib to runtime links, improved service ordering/synchronization, and streamlined package wiring.
  * Added HTTP mock server integration with lifecycle and port propagation.

* **New Features**
  * Added an end-to-end PostGIS multi-version upgrade test.

* **Bug Fixes**
  * Improved mock-server port-file directory handling, logging, and error propagation.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jfroche committed:
feat: enable nixosTest on aarch64 darwin

Build nixosTest using a local aarch64 linux builder and run nixosTest qemu VM in
aarch64 darwin. This keeps the tests fast as they don't require nested virtualisation.

For example, you can run the nixos test for the http extension on your aarch64 macOS with:

```shell
nix build .\#checks.aarch64-darwin.ext-http
```

jfroche committed:
chore: reduce warnings related to system

Current system value has been moved into `pkgs.stdenv.hostPlatform` by https://github.com/NixOS/nixpkgs/pull/456527

jfroche committed:
fix: remove redundant postgresqlWithExtension wrapper in extension tests

The tests were unnecessarily wrapping psql_15 with postgresqlWithExtension when psql_15 already includes the extensions.

jfroche committed:
fix: remove port forwarding from test configurations

This cause problem starting nixos tests on darwin hosts.

yvan-sraka committed:
fix: remaining nixos test references to use pkgs.pkgsLinux

- self.legacyPackages.${pkgs.stdenv.hostPlatform.system} -> pkgs.pkgsLinux.stdenv.hostPlatform.system
- self.packages.${pkgs.stdenv.hostPlatform.system} -> pkgs.pkgsLinux.stdenv.hostPlatform.system

samrose commented on nix/ext/tests/http.nix:
```diff
@@ -89,24 +86,40 @@ self.inputs.nixpkgs.lib.nixos.runTest {
+          mkdir -p /tmp
+
+          # Start the mock server
+          exec ${pkgs.python3}/bin/python3 ${mockServer}
```
```suggestion
          exec ${pkgs.pkgsLinux.python3}/bin/python3 ${mockServer}
```

yvan-sraka committed:
fix: check full PostGIS version instead of just major version

Changes the version assertion to compare the complete version string
rather than only the major version, providing more precise validation.

yvan-sraka committed:
fix: improve HTTP mock server error handling and directory creation

- Guard against empty dirname when creating port file directory
- Use bare 'raise' to preserve original exception traceback
- Only create directory if dirname is non-empty

steve-chavez commented:
> Build nixosTest using a local aarch64 linux builder and run nixosTest qemu VM in aarch64 darwin.

I wonder if anyone can explain how is the above accomplished? Like where the qemu VM is spawned on this PR? I don't see code related to that.

jfroche commented:
> > Build nixosTest using a local aarch64 linux builder and run nixosTest qemu VM in aarch64 darwin.
> 
> I wonder if anyone can explain how is the above accomplished? Like where the qemu VM is spawned on this PR? I don't see code related to that.

Hello @steve-chavez  ! We made another change in #1999 to simplify the process of running nixosTest on aarch64 darwin.
The related documentation is now merged in https://github.com/supabase/postgres/blob/6a6be67bef48c6d8c2866fbf266967fb755bb79c/nix/docs/nixos-tests-on-macos.md
Do not hesitate to test it out and give us feedback about it !
